### PR TITLE
Add Spellcasting and Bounded Spellcasting Archetypes Journals, Link Journal Entries to Spellcasting Archetype Feats

### PR DESCRIPTION
**Pull request by avagdumortesin**
Closes https://github.com/foundryvtt/pf2e/issues/11439

### CODE REVIEW

**Comment by avagdumortesin** [comment]
As the text was from the Remaster, I did not add it to all spellcasting archetypes, only the ones modified in _Player Core_.

File: packs/journals/archetypes.json [hunk 1]
```diff
@@ -3546,6 +3546,27 @@
...
..., they grant you a 2nd-rank spell slot, and if you have a spell repertoire, you can select one spell from your repertoire as a signature spell. At 8th level, they grant you a 3rd-rank spell slot. Archetypes refer to these benefits as the “basic spellcasting benefits.”</p><h2>Expert Spellcasting Feat</h2><p>Typically taken at 12th level, these feats make you an expert in spell attack rolls and DCs of the appropriate magical tradition and grant you a 4thrank spell slot. If you have a spell repertoire, you can select a second spell from your repertoire as a signature spell. At 14th level, they grant you a 5th-rank spell slot, and at 16th level, they grant you a 6th-rank spell slot. Archetypes refer to these benefits as the “expert spellcasting benefits.”</p><h2>Master Spellcasting Feat</h2><p>Usually found at 18th level, these feats make you a master in spell attack rolls and DCs of the appropriate magical tradition and grant you a 7thrank spell slot. If you have a spell repertoire, you can select a third spell from your repertoire as a signature spell. At 20th level, they grant you an 8th-rank spell slot. Archetypes refer to these benefits as the “master spellcasting benefits.”</p>",
```
Comment (TikaelSol) [inline packs/journals/archetypes.json:3558]:
```suggestion
                "content": "<p>Some archetypes grant you a substantial degree of spellcasting, albeit delayed compared to a character from a spellcasting class. A spellcasting archetype allows you to use scrolls, staves, and wands in the same way that a member of a spellcasting class can.</p><p>Spellcasting archetypes always grant the ability to cast cantrips in their dedication, and then they have a basic spellcasting feat, an expert spellcasting feat, and a master spellcasting feat. These feats share their name with the archetype; for instance, the druid's master spellcasting feat is called Master Druid Spellcasting. All spell slots you gain from spellcasting archetypes are subject to the restrictions within the archetype. For instance, the witch archetype allows you to pick a patron when you take its dedication feat. If you pick patron granting occult spells, the archetype then grants you spell slots you can use only to cast occult spells you prepare as a witch, even if you are a bard with occult spells in your repertoire.</p><h2>Basic Spellcasting Feat</h2><p>Usually available at 4th level, these feats grant a 1st-rank spell slot. At 6th level, they grant you a 2nd-rank spell slot, and if you have a spell repertoire, you can select one spell from your repertoire as a signature spell. At 8th level, they grant you a 3rd-rank spell slot. Archetypes refer to these benefits as the \"basic spellcasting benefits.\"</p><h2>Expert Spellcasting Feat</h2><p>Typically taken at 12th level, these feats make you an expert in spell attack rolls and DCs of the appropriate magical tradition and grant you a 4thrank spell slot. If you have a spell repertoire, you can select a second spell from your repertoire as a signature spell. At 14th level, they grant you a 5th-rank spell slot, and at 16th level, they grant you a 6th-rank spell slot. Archetypes refer to these benefits as the \"expert spellcasting benefits.\"</p><h2>Master Spellcasting Feat</h2><p>Usually found at 18th level, these feats make you a master in spell attack rolls and DCs of the appropriate magical tradition and grant you a 7thrank spell slot. If you have a spell repertoire, you can select a third spell from your repertoire as a signature spell. At 20th level, they grant you an 8th-rank spell slot. Archetypes refer to these benefits as the \"master spellcasting benefits.\"</p>",
```

Comment (TikaelSol) [reply packs/journals/archetypes.json:3558]: Well, it won't let me apply this for some reason, but get rid of the rsquo in "druid's" and replace the double quotes with standard `"` (escaping if editing the raw json).

File: packs/feats/basic-cleric-spellcasting.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. You can prepare your deity's spells in your spell slots of the appropriate level from the cleric archetype.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. You can prepare your deity's spells in your spell slots of the appropriate level from the cleric archetype.</p>"
```
Comment (stwlam) [inline packs/feats/basic-cleric-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. You can prepare your deity's spells in your spell slots of the appropriate rank from the cleric archetype.</p>"
```

File: packs/feats/basic-eldritch-archer-spellcasting.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of the appropriate spell level to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you have learned or discovered.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of the appropriate spell level to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you have learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-eldritch-archer-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of the appropriate spell rank to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you have learned or discovered.</p>"
```

File: packs/feats/basic-oracle-spellcasting.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. When you gain a spell slot of a new level from the oracle archetype, add a common divine spell or another divine spell you have learned or discovered to your repertoire, of that spell level.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. When you gain a spell slot of a new level from the oracle archetype, add a common divine spell or another divine spell you have learned or discovered to your repertoire, of that spell level.</p>"
```
Comment (stwlam) [inline packs/feats/basic-oracle-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. When you gain a spell slot of a new rank from the oracle archetype, add a common divine spell or another divine spell you have learned or discovered to your repertoire, of that spell level.</p>"
```

File: packs/feats/basic-psychic-spellcasting.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. When you gain a spell slot of a new level from the psychic archetype, add a spell of the appropriate spell level to your repertoire: a common occult spell, one of the granted spells from your conscious mind, or another spell you've learned or discovered.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. When you gain a spell slot of a new level from the psychic archetype, add a spell of the appropriate spell level to your repertoire: a common occult spell, one of the granted spells from your conscious mind, or another spell you've learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-psychic-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. When you gain a spell slot of a new level from the psychic archetype, add a spell of the appropriate spell rank to your repertoire: a common occult spell, one of the granted spells from your conscious mind, or another spell you've learned or discovered.</p>"
```

File: packs/feats/basic-sorcerer-spellcasting.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. Each time you gain a spell slot of a new level from the sorcerer archetype, add a spell of the appropriate spell level to your repertoire: a common spell of your bloodline's tradition, one of your bloodline's granted spells, or another spell you have learned or discovered.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from the sorcerer archetype, add a spell of the appropriate spell level to your repertoire: a common spell of your bloodline's tradition, one of your bloodline's granted spells, or another spell you have learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-sorcerer-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from the sorcerer archetype, add a spell of the appropriate spell rank to your repertoire: a common spell of your bloodline's tradition, one of your bloodline's granted spells, or another spell you have learned or discovered.</p>"
```

File: packs/feats/basic-spellcasting.json [hunk 7]
```diff
@@ -11,7 +11,7 @@
...
         "category": "class",
         "description": {
-            "value": "<p>You increase your ability to cast spells. You gain the basic spellcasting benefits granted by spellcasting archetypes, gaining a spell repertoire. Each time you gain a spell slot of a new level from the basic, expert, and master spellcasting benefits from pervasive magic feats, add a spell of the appropriate spell level to your repertoire: a common spell of the magical tradition matching your trait, or another spell of that tradition that you have learned or discovered.</p>"
+            "value": "<p>You increase your ability to cast spells. You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits granted by spellcasting archetypes, gaining a spell repertoire. Each time you gain a spell slot of a new level from the basic, expert, and master spellcasting benefits from pervasive magic feats, add a spell of the appropriate spell level to your repertoire: a common spell of the magical tradition matching your trait, or another spell of that tradition that you have learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-spellcasting.json:14]:
```suggestion
            "value": "<p>You increase your ability to cast spells. You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits granted by spellcasting archetypes, gaining a spell repertoire. Each time you gain a spell slot of a new level from the basic, expert, and master spellcasting benefits from pervasive magic feats, add a spell of the appropriate spell rank to your repertoire: a common spell of the magical tradition matching your trait, or another spell of that tradition that you have learned or discovered.</p>"
```

File: packs/feats/basic-beast-gunner-spellcasting.json [hunk 8]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You've unlocked the first secrets of beast guns, granting you the power of their synergistic magic. You gain the basic spellcasting benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of the appropriate spell level to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you've learned or discovered.</p>"
+            "value": "<p>You've unlocked the first secrets of beast guns, granting you the power of their synergistic magic. You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of the appropriate spell level to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you've learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-beast-gunner-spellcasting.json:14]:
```suggestion
            "value": "<p>You've unlocked the first secrets of beast guns, granting you the power of their synergistic magic. You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new rank from this archetype, add a spell of the appropriate spell rank to your repertoire, either a common spell of your chosen tradition or another spell of that tradition you've learned or discovered.</p>"
```

File: packs/feats/basic-cathartic-spellcasting.json [hunk 9]
```diff
@@ -11,7 +11,7 @@
         },
         "category": "class",
         "description": {
-            "value": "<p>You gain the basic spellcasting benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of that spell level from your chosen tradition to your repertoire-either a common spell or another spell you've learned or discovered.</p>"
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new level from this archetype, add a spell of that spell level from your chosen tradition to your repertoire-either a common spell or another spell you've learned or discovered.</p>"
```
Comment (stwlam) [inline packs/feats/basic-cathartic-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.DOc3Pf8wmVxanTIv#basic-spellcasting-feat]{Basic Spellcasting Feat} benefits. Each time you gain a spell slot of a new rank from this archetype, add a spell of that spell rank from your chosen tradition to your repertoire-either a common spell or another spell you've learned or discovered.</p>"
```

File: packs/feats/basic-summoner-spellcasting.json [hunk 10]
```diff
@@ -11,7 +11,7 @@
...
+            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.TwSQ0VPXZL0mfPez#basic-bounded-spellcasting-feat]{Basic Bounded Spellcasting Feat} benefits. You gain the Cast a Spell activity. Your key spellcasting ability for summoner archetype spells is Charisma, and they are summoner spells of your eidolon's tradition. You become trained in spell attack rolls and spell DCs of that tradition. You gain a spell repertoire and two cantrips. Each time you gain a spell slot of a new level from the summoner archetype, add a spell of that level or lower to your repertoire. Each time you lose spell slots of a particular level, remove those spells from your repertoire.</p>"
```
Comment (stwlam) [inline packs/feats/basic-summoner-spellcasting.json:14]:
```suggestion
            "value": "<p>You gain the @UUID[Compendium.pf2e.journals.JournalEntry.vx5FGEG34AxI2dow.JournalEntryPage.TwSQ0VPXZL0mfPez#basic-bounded-spellcasting-feat]{Basic Bounded Spellcasting Feat} benefits. You gain the Cast a Spell activity. Your key spellcasting ability for summoner archetype spells is Charisma, and they are summoner spells of your eidolon's tradition. You become trained in spell attack rolls and spell DCs of that tradition. You gain a spell repertoire and two cantrips. Each time you gain a spell slot of a new rank from the summoner archetype, add a spell of that rank or lower to your repertoire. Each time you lose spell slots of a particular rank, remove those spells from your repertoire.</p>"
```

**Pull request merged**